syntax = "proto3";

package DSConvoProtocol;

message DSConvoMessage {
    enum MessageType {
        INVALID = 0;
        HELLO_REQUEST = 1;
        HELLO_REPLY = 2;
        MESSAGE_REQUEST = 3;
        MESSAGE_BROADCAST = 4;
        GOODBYE = 5;
    }

    MessageType type = 1;
    bytes payload = 2;
}

message HelloRequestPayload {
    string username = 1;
}

message HelloReplyPayload {
    enum HelloReplyError {
        INVALID = 0;
        OK = 1;
        BAD_USERNAME = 2;
        USERNAME_IN_USE = 3;
        SERVER_LOAD = 4;
    }

    HelloReplyError result = 1;
    string username = 2;
}

message MessageRequestPayload {
    string msg = 1;
}

message MessageBroadcastPayload {
    string username = 1;
    string display_name = 2;
    string msg = 3;
}

// vim: set ts=4 sts=4 sw=4 et cin:
